server/src/main/java/io/crate/execution/engine/indexing/ShardingUpsertExecutor.java

```diff
@@ -365,9 +367,10 @@ private void countdown() {
             }
         }
 
-        private void maybeSetInterrupt(@Nullable Exception failure) {
-            if (failure instanceof InterruptedException) {
-                interrupt.set(failure);
+        private void maybeSetFailure(@Nullable Exception failure) {
+            if (failure != null) {
+                Throwable t = SQLExceptions.unwrap(failure);
+                interrupt.set(Exceptions.toRuntimeException(t));
             }
```

Comment:
On the contrary to `maybeRaiseFailure`, which affects UPDATE code path,

we don't need here handling of `DocumentMissingException` and `VersionConflictEngineException` as they are irrelevant for INSERT.